### PR TITLE
Fixes #280. Add non-normative vulnerability information extracted fro…

### LINKED ISSUE #280: Advisory's vulnerabilities do not have "title", "published", "modified" fields populated
- The endpoint `GET api/v1/advisory/{sha256}` returns and advisory, including its vulnerabilities. The vulnerabilities do not have the fields described in the title of this issue.
- This is an example of the current response generated by the backend
```json
{
    "identifier": "RHBA-2022:1352",
    "sha256": "f731216d1e36e974faffb05e55a585da0297a2a1ebff56d98dee5a884a9bc067",
    "issuer": "Red Hat Product Security",
    "published": "2022-04-13T14:30:05Z",
    "modified": "2024-05-02T19:23:44Z",
    "title": "Red Hat Bug Fix Advisory: .NET Core 3.1 on RHEL 7 bugfix update",
    "vulnerabilities": [
        {
            "identifier": "CVE-2022-0613",
            "title": null,
            "published": null,
            "modified": null,
            "withdrawn": null,
            "severity": "none",
            "score": 0.0,
            "cvss3_scores": [],
            "assertions": {}
        }
    ]
}
```

- The required fields will serve the population of the columns in the table below (UI screenshot)

Screenshot from 2024-05-14 10-30-11

### PR DESCRIPTION
**Pull request by bobmcwhirter**
…m advisories.

### CODE REVIEW

File: migration/src/m0000280_add_advisory_vulnerability_meta.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .drop_table(
+                Table::drop()
+                    .table(AdvisoryVulnerability::Table)
+                    .if_exists()
+                    .to_owned(),
+            )
```
Comment (jcrossley3) [inline migration/src/m0000280_add_advisory_vulnerability_meta.rs:37]: Shouldn't these be `drop_column`'s?

File: modules/fundamental/src/advisory/model/details/advisory_vulnerability.rs [hunk 2]
```diff
@@ -283,3 +312,19 @@ impl AdvisoryVulnerabilitySummary {
         Ok(summaries)
     }
 }
+
+#[derive(Serialize, Deserialize, Debug, Clone, ToSchema)]
+pub struct NonNormativeVulnerabilityInfo {
```
Comment (jcrossley3) [inline modules/fundamental/src/advisory/model/details/advisory_vulnerability.rs:317]: This smells a little like our old `NonLeaking...` types. Wouldn't both a normative and a non-normative info struct have the same fields?

**Review by jcrossley3 — COMMENTED** [review]
Forgive my ignorance, but I was thinking this fix would've allowed me to uncomment [these lines](https://github.com/trustification/trustify/pull/340/files#diff-c33d1be4aaa878cadebd411139a5caaca03eabdeff42d1e7d5a0bdf250ccde6aR219-R249), too

File: migration/src/m0000280_add_advisory_vulnerability_meta.rs (hunk 1 above)
Comment (bobmcwhirter) [reply migration/src/m0000280_add_advisory_vulnerability_meta.rs:37]: Yes thanks!

File: modules/fundamental/src/advisory/model/details/advisory_vulnerability.rs (hunk 2 above)
Comment (bobmcwhirter) [reply modules/fundamental/src/advisory/model/details/advisory_vulnerability.rs:317]: This is intentional. I'll document in-code.

**Review by jcrossley3 — APPROVED** [review]

**Pull request merged**